documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -0,0 +1,223 @@
+  - Enables S2S users to access underlying analytics data via API.
+  - Allows us to include narrative text that explains the metrics we publish.
+  - Enables open source contributors to host local versions of our dashboard or easily create their own dashboards.
+  - Aligns with the open source values and approach of the project.
+  - Data endpoints are simple and easy to understand.
+- **Cons**
+  - Building or modifying dashboards is quite technical, not something a business analyst could do on their own.
+  - Publishing new charts or dashboards requires making changes both to the API and to the dashboard UI separately.
+  - Hardest to maintain of all of the options considered.
+  - Does not easily support adhoc reports or dashboards without using a separate tool.
+
+### Custom dashboard app
+
+This option involves building a custom dashboard app using an existing framework, such as [Dash][plotly-dash] (python) or [R Shiny][r-shiny]. This custom built application would be hosted as its own application and be slightly more integrated than a separate analytics API + dashboard UI, but it would be more customizable than using an open source or proprietary dashboard solution.
+
+> [!TIP]
+> **Bottom line:** This option would be best if:
+> - we want to retain control over the look and feel of the dashboard *and* to manage the backend and frontend of the dashboard together,
+> - but we are willing to commit more engineering resources to building and hosting it *and* don't need to expose analytics data via API.
+
+- **Pros**
+  - Provides more fine-grained control over the look and feel of the dashboard application, on par with option 1.
+  - Allows a single engineer familiar with Python (or R if we use Shiny) to manage both the backend (e.g. analytics) and frontend (visualization and design of the dashboard).
+  - Faster to implement new charts and dashboards than an analytics API and separate frontend.
+  - Enables us to combine multiple dashboards into a single application.
+  - Allows open source contributors to host local versions of our dashboards.
+  - Aligns with the open source values and approach of the project.
+- **Cons**
+  - Building or modifying dashboards is still quite technical, not something a business analyst could do on their own.
+  - Harder to implement new charts or dashboards than SaaS or open source dashboard solution.
+  - Does not easily support adhoc reports or dashboards in the same tool.
+  - Does not easily support exposing analytics data to S2S users via API.
+
+### Open source dashboard solution
+
+This option involves selecting and hosting an open source dashboard solution, such as [Metabase][metabase] or [Redash][redash]. This solution would most likely be self-hosted and connect directly to our data warehouse, and enable business analysts to build adhoc reporting and dashboards with SQL and a drag-and-drop interface. Individual dashboards can then be configured for broader publication to external stakeholders.
```

Comment:
I've never heard of `redesh` or `metabase`, so I'm like coming at this from a different technological POV. That said, I think [Grafana](https://grafana.com/) is a strong choice here. Especially now that there's an 1st class [Amazon managed option](https://aws.amazon.com/grafana/). 

The major con is Grafana is that it's built for engineers rather than analysts, so there's a higher learning curve on the query language.

Reply:
Actually! I might redact this suggestion. Everything else on this list is a business insights dashboard, and Grafana is certainly not build for business insights.

Reply:
I like the idea of including Grafana in a future ADR that is focused on selecting _which_ open source dashboard tool we might want to use.

But agree that Grafana is more focused on observability and operational metrics in the traditional meaning of the word (e.g. load times, CPU usage, memory availability, telemetry) while Redash, Metabase, and Apache Superset are more accurately characterized as BI tools.

Reply:
Ah, apologies. I'm just now reading the part where you say that this ADR isn't for choosing tools!